runtime/src/utils/buffer/append.rs

```diff
@@ -0,0 +1,395 @@
+    /// the buffer pool.
+    async fn flush(&self, buffer: &mut Buffer) -> Result<(), Error> {
+        // Take the buffered data, if any.
+        let Some((buf, offset)) = buffer.take() else {
+            return Ok(());
+        };
+
+        // Insert the flushed data into the buffer pool. This step isn't just to ensure recently
+        // written data remains cached for future reads, but is in fact required to purge
+        // potentially stale cache data which might result from the edge the case of rewinding a
+        // blob across a page boundary.
+        let remaining = self.pool_ref.cache(self.id, &buf, offset).await;
+
+        // If there's any data left over that doesn't constitute an entire page, re-buffer it into
+        // the append buffer to maintain its page-boundary alignment.
+        if remaining != 0 {
+            buffer.offset -= remaining as u64;
+            buffer.data.extend_from_slice(&buf[buf.len() - remaining..])
+        }
+
+        // Write the data buffer to the underlying blob. Note that the implementation will rewrite
+        // the last (blob_size % page_size) "trailing bytes" of the underlying blob since the
+        // write's starting offset is always page aligned. We don't expect this inefficiency to be a
+        // significant performance concern, but would be easy enough to avoid by maintaining the
+        // underlying blob's size.
+        self.blob.write_at(buf, offset).await?;
+
+        Ok(())
+    }
+
+    /// Clones and returns the underlying blob.
+    pub fn clone_blob(&self) -> B {
+        self.blob.clone()
+    }
+}
+
+impl<B: Blob> Blob for Append<B> {
+    async fn read_at(
+        &self,
+        buf: impl Into<StableBuf> + Send,
+        offset: u64,
+    ) -> Result<StableBuf, Error> {
+        // Prepare `buf` to capture the read data.
+        let mut buf = buf.into();
+
+        // Ensure the read doesn't overflow.
+        let end_offset = offset
+            .checked_add(buf.len() as u64)
+            .ok_or(Error::OffsetOverflow)?;
+
+        // Acquire a read lock on the buffer.
+        let buffer = self.buffer.read().await;
+
+        // If the data required is beyond the size of the blob, return an error.
+        if end_offset > buffer.size() {
+            return Err(Error::BlobInsufficientLength);
+        }
+
+        // Extract any bytes from the buffer that overlap with the requested range.
+        let remaining = buffer.extract(buf.as_mut(), offset);
+        if remaining == 0 {
+            return Ok(buf);
+        }
+
+        // If there are bytes remaining to be read, use the buffer pool to get them.
+        self.pool_ref
+            .read(&self.blob, self.id, &mut buf.as_mut()[..remaining], offset)
+            .await?;
+
+        Ok(buf)
+    }
+
+    /// This [Blob] trait method is unimplemented by [Append] and unconditionally panics.
+    ///
+    /// TODO(<https://github.com/commonwarexyz/monorepo/issues/1207>): Extend the buffer pool to
```

Comment:
nit: In other areas, we leave TODOs with `//` (don't need to show up in docs IMO)

Reply:
Moved comment and prefixed with `//`